eng/build.yml

```diff
@@ -40,22 +40,35 @@ jobs:
       workspace:
         clean: all
       variables:
-        - ${{ insert }}: ${{ parameters.variables }}
-        - _AgentOSName: ${{ parameters.agentOs }}
-        - _TeamName: DotNetCore
-        - _OfficialBuildIdArgs: ''
-        - _PublishArgs: ''
-        - _SignArgs: ''
-        - _HelixApiToken: ''
+        - ${{ each variable in parameters.variables }}:
+          - ${{ if ne(variable.name, '') }}:
+            - name: ${{ variable.name }}
+              value: ${{ variable.value }}
+        - name: _AgentOSName
+          value: ${{ parameters.agentOs }}
+        - name: _TeamName
+          value: DotNetCore
+        - name: _OfficialBuildIdArgs
+          value: ''
+        - name: _PublishArgs
```

Comment:
I think you're overriding the publish args with an empty set in the official build.